Ladybird/AppKit/Application/ApplicationBridge.cpp

```diff
@@ -37,13 +39,47 @@ ErrorOr<void> ApplicationBridge::launch_request_server(Vector<ByteString> const&
     return {};
 }
 
+static ErrorOr<NonnullRefPtr<ImageDecoderClient::Client>> launch_new_image_decoder()
+{
+    auto image_decoder_paths = TRY(get_paths_for_helper_process("ImageDecoder"sv));
+    return launch_image_decoder_process(image_decoder_paths);
+}
+
+ErrorOr<void> ApplicationBridge::launch_image_decoder()
+{
+    m_impl->image_decoder_client = TRY(launch_new_image_decoder());
+
+    m_impl->image_decoder_client->on_death = [this] {
+        m_impl->image_decoder_client = nullptr;
+        if (auto err = this->launch_image_decoder(); err.is_error()) {
+            dbgln("Failed to restart image decoder: {}", err.error());
+            VERIFY_NOT_REACHED();
+        }
+
+        auto num_clients = WebView::WebContentClient::client_count();
+        auto new_sockets = m_impl->image_decoder_client->send_sync_but_allow_failure<Messages::ImageDecoderServer::ConnectNewClients>(num_clients);
+        if (!new_sockets || new_sockets->sockets().size() == 0) {
+            dbgln("Failed to connect {} new clients to ImageDecoder", num_clients);
```

Comment:
This will also fail if we have more than ... 64? tabs at once, since we can't transfer more sockets than that in one shot. But tab hoarders are silly people anyway. Even if Tim Cook is in that category :P